### ISSUE TITLE
How AdaIN works?

### ISSUE THREAD

**Issue by mkstmyk** [soomean/Diverse-Motion-Stylization#5]
Hi again! Please allow me to ask a newbie question.
The way of calculating the mean and variance in AdaIN is a bit tricky for me.
It seems to be just using nn.Linear to compute them at once and split it into gamma(variance) and beta(mean).
Will it lean to compute mean and variance during the training?
If so, where can I find the process to guarantee it?
Thank you again in advance.

`class AdaIN(nn.Module):`

    def __init__(self, style_dim, num_features):
        super(AdaIN, self).__init__()
        self.norm = nn.InstanceNorm2d(num_features, affine=False)
        self.fc = nn.Linear(style_dim, num_features * 2)

    def forward(self, x, s):
        h = self.fc(s)
        h = h.view(h.size(0), h.size(1), 1, 1)
        gamma, beta = torch.chunk(h, chunks=2, dim=1)
        return (1 + gamma) * self.norm(x) + beta
